sdk/src/main/java/io/opentelemetry/sdk/metrics/AggregatorMap.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.LabelSet;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.aggregator.Aggregator;
+import io.opentelemetry.sdk.metrics.aggregator.AggregatorFactory;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+abstract class AggregatorMap {
+  static AggregatorMap getNoop() {
+    return Noop.INSTANCE;
+  }
+
+  static AggregatorMap getAllLabels(
+      Descriptor descriptor,
+      Resource resource,
+      InstrumentationLibraryInfo instrumentationLibraryInfo,
+      AggregatorFactory aggregatorFactory,
+      long startEpochNanos) {
+    return new AllLabels(
+        descriptor, resource, instrumentationLibraryInfo, aggregatorFactory, startEpochNanos);
+  }
+
+  abstract void collect(LabelSet labelSet, Aggregator aggregator);
+
+  abstract List<MetricData> stopCollection(long epochNanos);
+
+  private static final class Noop extends AggregatorMap {
+    private static final AggregatorMap INSTANCE = new Noop();
+
+    @Override
+    void collect(LabelSet labelSet, Aggregator aggregator) {
+      // Noop.
+    }
+
+    @Override
+    List<MetricData> stopCollection(long epochNanos) {
+      return Collections.emptyList();
+    }
+  }
+
+  private static final class AllLabels extends AggregatorMap {
+    private final Descriptor descriptor;
+    private final Resource resource;
+    private final InstrumentationLibraryInfo instrumentationLibraryInfo;
+    private final AggregatorFactory aggregatorFactory;
+    private final Map<Map<String, String>, Aggregator> aggregatorMap;
```

Comment:
I think we can use `LabelSetSdk` as the key to this map, since it has a well-defined equals/hashcode via autovalue.